raiden_contracts/contracts/TokenNetwork.sol

```diff
@@ -693,28 +707,29 @@ contract TokenNetwork is Utils {
     /// the locked tokens corresponding to locks with secrets registered on-chain to the
     /// `participant`. Locked tokens corresponding to locks where the secret was not revelead
     /// on-chain will return to the `partner`. Anyone can call unlock.
+    /// @param channel_identifier Identifier for the channel on which this operation takes place.
     /// @param participant Address who will receive the unlocked tokens.
     /// @param partner Address who sent the pending transfers.
     /// @param merkle_tree_leaves The entire merkle tree of pending transfers that `partner`
     /// sent to `participant`.
     function unlock(
+        uint256 channel_identifier,
         address participant,
         address partner,
         bytes merkle_tree_leaves
     )
         public
     {
+        // Channel represented by channel_identifier must be settled and channel data deleted
+        require(channel_identifier != getChannelIdentifier(participant, partner));
```

Comment:
This is essentially irrelevant for the current `unlock` implementation. `channel_identifier` can be any value that the client sets. We don't currently keep the `channel_identifier` in storage at this point in the channel state.
It is nice to have the `channel_identifier` in the `ChannelUnlock` event for the Raiden client, as discussed.

Therefore, we can continue the `unlock` discussion here: https://github.com/raiden-network/raiden-contracts/issues/202